pkg/archive/paths.go

```diff
@@ -77,31 +123,45 @@ func WalkTarGzFiles(ctx context.Context, path string, walkFn FileWalkFn) (err er
 			err = cErr
 		}
 	}()
-
-	gzipReader, err := gzip.NewReader(file)
-
+	reader, err := getTarReader(file)
 	if err != nil {
 		return err
 	}
-	tarReader := tar.NewReader(gzipReader)
+
 	for {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
 		default:
 		}
-		header, err := tarReader.Next()
+		header, err := reader.Next()
 		if err == io.EOF {
 			break
 		}
 		if err != nil {
 			return err
 		}
-		if proceed, err := walkFn(ctx, header.Name, header.Size, tarReader); err != nil {
+		if proceed, err := walkFn(ctx, header.Name, header.Size, reader); err != nil {
 			return err
 		} else if !proceed {
 			break
 		}
 	}
 	return nil
+
+}
+
+func CheckArchiveType(filename string) (FormatType, bool) {
```

Comment:
Let's rename to something like `ParseFilenameFormatType`.
We aren't just checking something, we're parsing and created a value.
We also are creating a `FormatType` from a filename so it makes sense to include that in the name.
Alternatively, `FormatType` could be renamed to `ArchiveType`. Happy for you to make the call!